ipyvue/VueWidget.py

```diff
@@ -86,18 +86,27 @@ def on_event(self, event_and_modifiers, callback, remove=False):
         ]:
             del self._event_handlers_map[existing_event]
 
-        self._event_handlers_map[event_and_modifiers] = CallbackDispatcher()
-
-        self._event_handlers_map[event_and_modifiers].register_callback(
-            callback, remove=remove
-        )
-
-        if remove and not self._event_handlers_map[event_and_modifiers].callbacks:
+        if isinstance(callback, str):
+            self._event_handlers_map[event_and_modifiers] = callback
+        else:
+            self._event_handlers_map[event_and_modifiers] = CallbackDispatcher()
+
+            self._event_handlers_map[event_and_modifiers].register_callback(
+                callback, remove=remove
```

Comment:
Slightly unrelated but I think this code (81-95) is very hard to read, or does not work. The register_callback call with remove=True is a no-op or maybe it gives an exception? 
Would make me feel more comfortable is we had a test for this.